File: fore-core-jv/src/main/java/co/early/fore/core/time/SystemTimeWrapper.java
package co.early.fore.core.time;

/**
 * This enables us to set a mock the system time for testing.
 */
public class SystemTimeWrapper {

    public long currentTimeMillis() {
        return System.currentTimeMillis();
    }

    public long nanoTime() {
        return System.nanoTime();
    }
}

File: fore-kt-core/src/main/java/co/early/fore/core/time/SystemTimeWrapper.kt
package co.early.fore.core.time

/**
 * This enables us to set a mock the system time for testing.
 */
class SystemTimeWrapper {
    fun currentTimeMillis(): Long {
        return System.currentTimeMillis()
    }

    fun nanoTime(): Long {
        return System.nanoTime()
    }
}